src/main/java/org/zalando/nakadi/service/subscription/state/ClosingState.java

```diff
@@ -39,9 +43,10 @@ public void onExit() {
 
     @Override
     public void onEnter() {
-        final long timeToWaitMillis = getParameters().commitTimeoutMillis - (System.currentTimeMillis()
-                - lastCommitMillis);
-        if (timeToWaitMillis > 0) {
+        final long timeToWaitMillis = getParameters().commitTimeoutMillis -
```

Comment:
It is not clear why use function here, but if it is a case then it is better to move this lines to constructor and make vars final.